presto-orc/src/main/java/com/facebook/presto/orc/reader/BooleanSelectiveStreamReader.java

```diff
@@ -0,0 +1,332 @@
+                        if (outputRequired) {
+                            nulls[outputPositionCount] = true;
+                        }
+                        outputPositions[outputPositionCount] = position;
+                        outputPositionCount++;
+                    }
+                }
+                else {
+                    boolean value = dataStream.nextBit();
+                    if (filter.testBoolean(value)) {
+                        if (outputRequired) {
+                            values[outputPositionCount] = (byte) (value ? 1 : 0);
+                            if (recordNulls) {
+                                nulls[outputPositionCount] = false;
+                            }
+                        }
+                        outputPositions[outputPositionCount] = position;
+                        outputPositionCount++;
+                    }
+                }
+                streamPosition++;
+            }
+        }
+
+        readOffset = offset + streamPosition;
+        return outputPositionCount;
+    }
+
+    private void skip(int items)
+            throws IOException
+    {
+        if (dataStream == null) {
+            presentStream.skip(items);
+        }
+        else if (presentStream != null) {
+            int dataToSkip = presentStream.countBitsSet(items);
+            dataStream.skip(dataToSkip);
+        }
+        else {
+            dataStream.skip(items);
+        }
+    }
+
+    private void ensureValuesCapacity(int capacity, boolean recordNulls)
+    {
+        if (values == null || values.length < capacity) {
+            values = new byte[capacity];
+        }
+
+        if (recordNulls) {
+            if (nulls == null || nulls.length < capacity) {
+                nulls = new boolean[capacity];
+            }
+        }
+    }
+
+    private void ensureOutputPositionsCapacity(int capacity)
+    {
+        if (outputPositions == null || outputPositions.length < capacity) {
+            outputPositions = new int[capacity];
+        }
+    }
+
+    @Override
+    public int[] getReadPositions()
+    {
+        return outputPositions;
+    }
+
+    @Override
+    public Block getBlock(int[] positions, int positionCount)
+    {
+        checkArgument(outputPositionCount > 0, "outputPositionCount must be greater than zero");
+        checkState(outputRequired, "This stream reader doesn't produce output");
+        checkState(positionCount <= outputPositionCount, "Not enough values");
+
+        if (positionCount == outputPositionCount) {
+            if (allNulls) {
+                return new RunLengthEncodedBlock(BOOLEAN.createBlockBuilder(null, 1).appendNull().build(), positionCount);
+            }
+
+            Block block = new ByteArrayBlock(positionCount, Optional.ofNullable(nulls), values);
+            nulls = null;
+            values = null;
+            return block;
+        }
+
+        byte[] values = new byte[positionCount];
+        boolean[] nulls = null;
```

Comment:
It's confusing to give this the same name as the member variable

Reply:
How about `valuesCopy` and `nullsCopy`?

Reply:
OK

Reply:
@tdcmeehan Renamed to `xxxCopy`.